### PR TITLE
Support `.edit {fileName|None}` in CLI

### LINKED ISSUE #870: Support `.edit {fileName|None}` in CLI
### Set editor type to environment variable `$EDITOR`
```
$> export EDITOR=vi
```
### `.edit fileName`
``` 
gluesql> .edit foo.sql
```
=> open Editor as executing `vi foo.sql`

### `.edit`
```
gluesql> SELECT 1, true, 'Previous SQL';
gluesql> .edit 
```
```
SELECT 1, true, 'Previous SQL';
```
=> open Editor and show previous SQL on temporary file like `/tmp/hashedRandomName.sql`

### `:wq`
should return to previous CLI screen right after finishing EDITOR

### PR DESCRIPTION
**Pull request by devgony**
To resolve #870 

## Implement
```
gluesql> .run
[error] Nothing in SQL history to run.
gluesql> SELECT 1, 'a', true
```
| 1 | 'a' | true |
|---|-----|------|
| 1 | a   | TRUE |
```
.edit
```
=> open Editor with last command on temporary file like /tmp/Glue_****.sql
```sql
-- modify in editor
SELECT 1, 'a', true, 2, 'b'
```
```
gluesql> .run
```
| 1 | 'a' | true | 2 | 'b' |
|---|-----|------|---|-----|
| 1 | a   | TRUE | 2 | b   |
```
gluesql> .edit foo.sql
```
```
-- modify in editor
SELECT 1 AS no, 'In physical file' AS name
```
```
gluesql> .execute foo.sql
```
| no | name             |
|----|------------------|
| 1  | In physical file |

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -124,12 +127,53 @@ where
...
+    fn execute(&mut self, sql: String) -> Result<()> {
+        match self.glue.execute(sql.as_str()) {
+            Ok(payloads) => self.print.payloads(&payloads)?,
+            Err(e) => {
+                println!("[error] {}\n", e);
+            }
+        };
+
+        Ok(())
+    }
```
Comment (ever0de) [inline cli/src/cli.rs:175]:
```suggestion
    fn execute(&mut self, sql: impl AsRef<str>) -> Result<()> {
        match self.glue.execute(sql) {
            Ok(payloads) => self.print.payloads(&payloads)?,
            Err(e) => {
                println!("[error] {}\n", e);
            }
        };

        Ok(())
    }
```

File: cli/src/cli.rs [hunk 2]
```diff
@@ -124,12 +127,53 @@ where
...
+                            let edited = edit_with_builder(last, &builder)?;
+                            rl.add_history_entry(edited);
+                        }
+                    };
+                }
+                Command::Run => {
+                    let sql = rl
+                        .history()
+                        .last()
+                        .map_or_else(|| Err(CommandError::LackOfSQLHistory), |v| Ok(v.to_owned()));
```
Comment (ever0de) [inline cli/src/cli.rs:149]:
```suggestion
                    let sql = rl.history().last().ok_or(CommandError::LackOfSQLHistory);
```

**Review by ever0de — APPROVED** [review]
Looks good 👍

**Review by panarch — APPROVED** [review]
Awesome!
`.edit` makes users be able to use the CLI program seriously.
Wow.. I didn't know this kind of features before.
Can't wait to release these new powerful CLI functions to the next version v0.13!
Thanks really a lot 👍 👍

**Pull request merged**